### ISSUE TITLE
Server Content Architecture Discussion

### ISSUE THREAD

**Issue by ejaxon** [CodeForNC/reentry-resources-hub#10]
How do we organize the information on the server to facilitate creating, customizing and maintaining local reentry sites.

The current plan is to store information in files rather than in a database. This has a couple major advantages:

- The entire state can be checked into Github. Note that we can allow admins to update the site via a client app, but then issue a periodic pull request (perhaps nightly) to re-merge into the main repo.
- We can begin hand-creating content and building the public site without waiting for an administrative client to be built.

It's pretty clear that we should build this as a country/state/county hierarchy (I expect regional resources should be designed via a region tag and automatically pulled together). Each level can have an intro file and a set of resources - not sure whether resources should be stored one-per-file or in a JSON array in a single file.

Also, do we want to offer the ability to control order of resources taken from a higher level (or even suppression)?

**Comment by marshallj** [comment]
Isn't it also cheaper to store the info in files since we don't need a database instance in the cloud? 

Also, @ejaxon do you have some sort of standard template to collect info from the folks running the services (community stakeholders)? If not should we put one together? Could be emailed to, or maybe a google form. Thinking it could help us organize and collect the info on services in a standardized way once we are ready to scale this to other communities. Also thinking it could help us define the regional services in an organized way.

**Comment by ejaxon** [comment]
Re: file storage, I agree - this approach lets us keep server resource requirements (and configuration pain) to a minimum. My current thinking is that we can create subdomains for each of the counties (e.g., guilford.ncreentryhub.org), but run it all off a single server that doesn't even need to be that powerful (we could actually run it off github pages, but probably not if we do the subdomains).

One other cost consideration: the custom google search, which is $100/year. Not sure if we can configure in such a way that a single instance works. If not, that gets pricey - something to investigate. It IS important given how many places out there are looking to scam the people we serve.

Re: template, I think that's a good idea. We should do that in collaboration with the local experts (and maybe United Way) to make sure we're covering the right questions.

**Comment by ejaxon** [comment]
I've just checked into the [content-arch branch](https://github.com/CodeForNC/reentry-resources-hub/tree/content-arch) both the proposed structure for all 100 counties and a script (in the ````init```` subdirectory) that creates it. I think it will probably mostly be clear. The one thing that I want to highlight is that the ````resources_local.json```` file at the state (common) level is the set of parameterized links that will be used in every local jurisdiction, replacing state with 'nc' and jurisdiction with the county tag (I probably should make the parameters used there the same as what is used in the config.json files).

__Update:__ I've added ````common_jurisdiction```` to the state (common) level ````config.json```` and ````local_jurisdiction```` to the county ones and changed the parameterized local resources file to use those instead. Also switched to double curly brackets as the escape for variables in that file.

I'm still working on the structure - just realizing that I need common-level content for all the pages.

**Comment by ejaxon** [comment]
Revamped the structure a bit. I think the current form is what I propose.

This means that creating a _topic_ page for the site involves reading the corresponding common page for that _topic_ (in content/pages/_topic_), which gives the common (state and national) description and parameterized links plus the corresponding page data in the local jurisdiction (county) directory (in content/jurisdictions/_county_/_topic_).

**Comment by ejaxon** [comment]
@marshallj I have created the proposed architecture in the [content-arch branch](https://github.com/CodeForNC/reentry-resources-hub/tree/content-arch) (and of course have the ````init_content.js```` script to regenerate if we change).  I also started documenting it in the documentation directory (full documentation can be one of the tasks).

Let me know if you're ok with this for now and I'll close this issue.
